### PR TITLE
Contributing documentation

### PR DESCRIPTION
**Pull request by amelieled**
The purpose of this PR is to discuss the rules for contributing to the project. I only modified the CONTRIBUTING. md file to make explicit some rules that are still implicit. 

You tell me what's missing!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -41,12 +73,12 @@ Note that `ocamldoc` comments should take up as much space as possible,
...
+You can do the same thing for nested functions, but it's not mandatory.
 
+Pattern matching
+----------------
 
-Other conventions
------------------
-
-Pattern matching arrows should be aligned to improve readability, as in the
+Pattern matching arrows should be aligned to improve readability, especially in the case where almost all of the pattern matching can be seen, as in the
```
Comment (fblanqui) [inline CONTRIBUTING.md:81]: I am in favor of removing this paragraph.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -61,3 +93,78 @@ by two units (that is, four characters).
...
+else (* second case *)
+```
+or, if it is not possible to write the second case on one line:
+```OCaml
+if (* test *) then
+>> (* first case *)
+else 
+>> (* second case *)
+```
+It is not readable if the `test` cannot to be written on one line, so you must write the first line in each template on one line only.
```
Comment (fblanqui) [inline CONTRIBUTING.md:114]: I do not understand this sentence.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -61,3 +93,78 @@ by two units (that is, four characters).
...
+(** Description of the inductive type*)
+type name =
+  { constructor1 : type1
+  (** Description of the field 1  *)
+         ...
+  ; constructorN : typeN
+  (** Description of the field N  *) }
+```
+
+Do the same thing with sum-type.
```
Comment (fblanqui) [inline CONTRIBUTING.md:138]: sum types

**Comment by fblanqui** [comment]
@amelieled when you do a PR, you should start from the master branch so that you have a clear history and it can be merged easily. Perhaps you want to close this PR and create a new one.

**Comment by amelieled** [comment]
Yes but I can't do that because of PR about induction.

**Comment by ejgallego** [comment]
`git rebase -i` should be easy.

**Comment by fblanqui** [comment]
In this case, as there is only one file to modify, there is may be a simpler solution: from origin/master, create a new branch doc with your modified file and create a new PR.

**Comment by ejgallego** [comment]
> In this case, as there is only one file to modify, there is may be a simpler solution: from origin/master, create a new branch doc with your modified file and create a new PR.

This can be done more efficiently by resetting this branch, the cherry-picking, then pushing again.

**Comment by ejgallego** [comment]
> This can be done more efficiently by resetting this branch, the cherry-picking, then pushing again.

Or just doing `git rebase master -i` and removing the unneeded commits.

**Pull request closed without merging**